spark/src/main/kotlin/com/adevinta/spark/components/textfields/SparkTextField.kt

```diff
@@ -0,0 +1,376 @@
+                    colors = colors,
+                    placeholder = { PlaceHolder(text = placeholder) },
+                    supportingText = supportingTextComposable,
+                    counter = counterComposable,
+                    leadingIcon = leadingIcon,
+                    trailingIcon = trailingIcon,
+                    singleLine = singleLine,
+                    enabled = enabled,
+                    isError = isError,
+                ) {
+                    OutlinedBorderContainerBox(
+                        enabled,
+                        isError,
+                        interactionSource,
+                        colors,
+                    )
+                }
+            },
+            onTextLayout = {},
+        )
+    }
+}
+
+/**
+ * Composable that draws a default container for [SparkTextField] with a border stroke. You
+ * can use it to draw a border stroke in your custom text field based on
+ * [OutlinedBorderContainerBox]. The [SparkTextField] applies it automatically.
+ *
+ * @param enabled whether the text field is enabled
+ * @param isError whether the text field's current value is in error
+ * @param interactionSource the [InteractionSource] of this text field. Helps to determine if
+ * the text field is in focus or not
+ * @param colors [TextFieldColors] used to resolve colors of the text field
+ */
+@ExperimentalMaterial3Api
+@Composable
+private fun OutlinedBorderContainerBox(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    colors: DefaultSparkTextFieldColors,
+) {
+    val shape = SparkTheme.shapes.extraSmall
+    val borderStroke = animateBorderStrokeAsState(
+        enabled,
+        isError,
+        interactionSource,
+        colors,
+    )
+    Box(
+        Modifier
+            .border(borderStroke.value, shape)
+            .background(colors.containerColor().value, shape),
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun animateBorderStrokeAsState(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    colors: DefaultSparkTextFieldColors,
+): State<BorderStroke> {
+    val focused by interactionSource.collectIsFocusedAsState()
+    val indicatorColor = colors.indicatorColor(enabled, isError, interactionSource)
+    val targetThickness =
+        if (focused) TextFieldDefaults.FocusedBorderThickness else TextFieldDefaults.UnfocusedBorderThickness
+    val animatedThickness = if (enabled) {
+        animateDpAsState(targetThickness, tween(durationMillis = 150))
+    } else {
+        rememberUpdatedState(TextFieldDefaults.UnfocusedBorderThickness)
+    }
+    return rememberUpdatedState(
+        BorderStroke(animatedThickness.value, SolidColor(indicatorColor.value)),
+    )
+}
+
+@Stable
+data class TextFieldCharacterCounter(
+    val count: Int,
+    val maxCharacter: Int,
+)
+
+@Composable
+private fun Label(text: String?) {
+    if (text?.take(30) != null) {
```

Comment:
This creates unnecessary strings. This in fact only checks if `text` is not null.